feat: Separate plan and provision jobs for simple agent registration

kylecarbs opened the pull request:
This is prep-work for integration of the workspace agent into the provision flow.

kylecarbs committed:
feat: Add agent authentication based on instance ID

Each cloud has it's own unique instance identity signatures, which
can be used for zero-token authentication. This change adds support
for tracking by "instance_id", and automatically authenticating
with Google Cloud.

kylecarbs committed:
Add test for CLI

kylecarbs committed:
Merge branch 'main' into agent

kylecarbs committed:
Add agent command

kylecarbs committed:
Separate plan and provision jobs for resource simplicity

kylecarbs committed:
Merge branch 'main' into agentauth

bryphe-coder reviewed (approved):
Looks good to me overall! Just some minor nits around logging in a few cases.

I also noticed there are a few places that are commented out - maybe it's worth removing those and re-introducing in a separate PR, if they can be split out?

kylecarbs committed:
Add agent authentication token generation

kylecarbs committed:
Make progress removing agent token

kylecarbs committed:
Add workspace resource middleware

kylecarbs committed:
Add peerbroker.proxy

kylecarbs committed:
feat: Add peerbroker proxy for agent connections

Agents will connect using this proxy. Eventually we'll intercept
some of these messages for validation, but that's not necessary right now.

kylecarbs committed:
Rename workspace agent auth

kylecarbs committed:
Fix linting errors

kylecarbs committed:
Merge branch 'peerproxy' into agentauth

kylecarbs committed:
Fix PostgreSQL UTF-8 pubsub

kylecarbs committed:
Fix data race with error

kylecarbs committed:
Show agent in CLI output

kylecarbs committed:
Add stop command
